rustls/src/session.rs

```diff
@@ -456,6 +456,19 @@ pub struct SessionCommon {
     pub(crate) quic: Quic,
 }
 
+/// Return true if the provided io::Error is the result of receiving a close_notify
+/// alert from the peer. Users should treat this as a clean shutdown of the connection.
```

Comment:
It seems wrong for us to treat `close_notify` as an error when really it's the least erroneous result we could get; only when we receive a `close_notify` has everything gone according to how the spec says it should. So, like I said in the related issue, it seems like we should redo the API to make receiving a close_notify an explicit non-error result.